go.mod

```diff
@@ -1,6 +1,6 @@
 module github.com/hyperledger/fabric
 
-go 1.23
+go 1.23.1
```

Comment:
> Why didn't it work to keep the 2-digit version? 1.23

It did work with the 2-digit version. However all the Go documentation says to use a 3-digit version for the toolchain. Since we don't have a different toolchain version listed, it will use the Go directive version instead. Therefore to be aligned with the documentation it is better to switch to the 3-digit version so that nothing unexpected happens. I realize I was reluctant to make this move earlier, but most of the Go community is moving to this approach and I'm now on board.